bodhi/server/consumers/masher.py

```diff
@@ -623,56 +687,106 @@ def remove_pending_tags(self):
+        raise NotImplementedError
 
-        if not os.path.exists(self.my_comps_dir):
-            util.cmd(['git', 'clone', comps_url, self.my_comps_dir],
-                     os.path.dirname(self.my_comps_dir))
+    def create_pungi_config(self):
+        loader = jinja2.FileSystemLoader(searchpath=config.get('pungiconf.basepath'))
+        env = jinja2.Environment(loader=loader, autoescape=False)
 
-        util.cmd(['git', 'pull'], self.my_comps_dir)
-        util.cmd(['make'], self.my_comps_dir)
+        env.globals['id'] = self.id
+        env.globals['release'] = self.release
+        env.globals['request'] = self.request
+        env.globals['updates'] = self.updates
+
+        config_template = config.get(self.pungi_template_config_key)
+        template = env.get_template(config_template)
+
+        pungi_conf_dir = tempfile.mkdtemp(prefix='bodhi-pungi-%s-' % self.id)
+
+        with open(os.path.join(pungi_conf_dir, 'pungi.conf'), 'w') as conffile:
+            conffile.write(template.render())
+
+        self.copy_additional_pungi_files(pungi_conf_dir, env)
+
+        return pungi_conf_dir
 
     def mash(self):
-        if self.path in self.state['completed_repos']:
+        if self.path and self.path in self.state['completed_repos']:
             self.log.info('Skipping completed repo: %s', self.path)
             return
 
-        comps = os.path.join(self.my_comps_dir, 'comps-%s.xml' %
-                             self.release.branch)
-        previous = os.path.join(config.get('mash_stage_dir'), self.id)
 
-        mash_thread = MashThread(self.id, self.path, comps, previous, self.log)
-        mash_thread.start()
-        return mash_thread
+        previous = self.get_previous_compose()
+        config_dir = self.create_pungi_config()
+        config_file = os.path.join(config_dir, 'pungi.conf')
+        pungi_cmd = [config.get('pungiconf.cmd'),
+                     '--config', config_file,
+                     '--no-label',
+                     '--target-dir', self.mash_dir]
+        if previous:
+            pungi_cmd += ['--old-composes', previous]
+
+        self.log.info('Running the pungi command: %s', pungi_cmd)
+        mash_process = subprocess.Popen(pungi_cmd,
+                                        # Nope. No shell for you
+                                        shell=False,
+                                        # Should be useless, but just to set something predictable
+                                        cwd=self.mash_dir,
+                                        # Pungi will logs its stdout into pungi.global.log
+                                        stdout=DEVNULL,
+                                        # Stderr should also go to pungi.global.log if it starts
+                                        stderr=subprocess.PIPE,
+                                        # We will never have additional input
+                                        stdin=DEVNULL)
```

Comment:
I know /dev/null is special, but will there be a problem if multiple processess are given the same file handle to it? Should we make a call to ```open()``` for each process rather than opening one at the module-level?